code/centauri/algorithms/grandpa/prover/src/lib.rs

```diff
@@ -237,7 +236,7 @@ where
 			.ok_or_else(|| anyhow!("Failed to fetch previous finalized hash + 1"))?;
 
 		let mut unknown_headers = vec![];
-		for height in previous_finalized_height..=latest_finalized_height {
+		for height in client_state.latest_relay_height..=latest_finalized_height {
```

Comment:
Why this change, could cause us to miss some parachain updates?

Reply:
Yes, exactly

Reply:
I mean the former code ensured we don’t miss any blocks

Reply:
What you changed it to is what he had there previously.

Reply:
David is correct, this code is wrong. Should be `previous_finalized_height`.

Reply:
Ok, I see what you mean